src/backend/api/objects.py

```diff
@@ -97,3 +143,464 @@ async def get_object_list(
         except s3.exceptions.ClientError:
             return None
     return res
+
+
+@router.head(
+    "/{bucket_name}/object",
+    responses=s3gw_client_responses(),
+)
+async def object_exists(
+    conn: S3GWClientDep, bucket_name: str, params: ObjectRequest
+) -> Response:
+    """
+    See
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/head_object.html
+    """
+    async with conn.conn() as s3:
+        await s3.head_object(
+            Bucket=bucket_name, Key=params.Key, VersionId=params.VersionId
+        )
```

Comment:
Can we add a comment here stating that we are relying on `head_object()` raising an Error in case the object does not exist?